src/agora/doer.h

```diff
@@ -22,22 +23,40 @@ class Doer {
       moodycamel::ProducerToken* worker_ptok) {
     EventData req_event;
 
-    ///Each event is handled by 1 Doer(Thread) and each tag is processed sequentually
-    if (task_queue.try_dequeue(req_event)) {
+    dequeue_start_tsc_ = GetTime::WorkerRdtsc();
+    auto ret = task_queue.try_dequeue(req_event);
+    dequeue_end_tsc_ = GetTime::WorkerRdtsc();
+    size_t dequeue_diff_tsc = dequeue_end_tsc_ - dequeue_start_tsc_;
+    dequeue_tsc_ = dequeue_diff_tsc;
+    /// Each event is handled by 1 Doer(Thread) and each tag is processed sequentually
+    if (ret) {
       // We will enqueue one response event containing results for all
       // request tags in the request event
       EventData resp_event;
       resp_event.num_tags_ = req_event.num_tags_;
       resp_event.event_type_ = req_event.event_type_;
 
       for (size_t i = 0; i < req_event.num_tags_; i++) {
+        if (req_event.event_type_ == EventType::kFFT) {
+          Packet* pkt = fft_req_tag_t(req_event.tags_.at(i)).rx_packet_->RawPacket();
+          frame_id_ = pkt->frame_id_;
+          symbol_id_ = pkt->symbol_id_;
+        } else {
+          // CAUTION: Proper value in symbol_id_ below is subject to the type of req_event
+          frame_id_ = gen_tag_t(req_event.tags_.at(i)).frame_id_;
+          symbol_id_ = gen_tag_t(req_event.tags_.at(i)).symbol_id_;
+        }
```

Comment:
Not sure if there is a better way to do this since symbol_id_  / frame_id_  could be different for new event types